oat-sa/tao-calculator-fe
# Feature/ADF-1417/Upgrade to Node 18

### Related to: [ADF-1417](https://oat-sa.atlassian.net/browse/ADF-1417) and [ADF-1419](https://oat-sa.atlassian.net/browse/ADF-1419)

Requires: 
 - [x] https://github.com/oat-sa/expr-eval/pull/16
 - [x] update the file [`package.json`](https://github.com/oat-sa/tao-calculator-fe/pull/33/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R52) once the companion PR has been merged and released

### Summary

Set Node 18 for the CI workflow.

Update the devDependencies accordingly.

### Details

To properly support Node 18, the dev dependencies had to be updated and aligned.
The plugin `terser` for rollup has been moved from the old package `rollup-plugin-terser` to `@rollup/plugin-terser`.

**Note:** The lock file was generated with `npm@6` so it would also work seamlessly with Node 14.

### How to test
- check the report on GitHub
- make sure to have Node 18 installed (you can use [nvm.sh](https://github.com/nvm-sh/nvm) for managing multiple versions)
- install the package: `npm i`
- run the unit tests: `npm test`, `npm run test:cov`
- test the bundling: `npm run build`
- test the linting: `npm run lint`
- test the sandbox: `npm run dev`


[ADF-1417]: https://oat-sa.atlassian.net/browse/ADF-1417?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ADF-1419]: https://oat-sa.atlassian.net/browse/ADF-1419?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
